### PR TITLE
Upload Images

### PR DESCRIPTION
**Pull request by Sjmarf**
You can now add links and images to posts

### CODE REVIEW

File: Mlem/App/Views/Pages/PostEditor/PostEditorView+Logic.swift [hunk 1]
```diff
@@ -13,10 +13,28 @@ extension PostEditorView {
...
+            && link == .none
+            && imageManager == nil
+    }
     
     var canSubmit: Bool {
-        !titleIsEmpty && targets.allSatisfy { $0.community != nil && $0.resolutionState == .success }
+        !titleIsEmpty
+            && targets.allSatisfy { $0.community != nil && $0.resolutionState == .success }
+            && link != .waiting
+            && (imageManager?.progress ?? 1) != 1
```
Comment (EricBAndrews) [inline Mlem/App/Views/Pages/PostEditor/PostEditorView+Logic.swift]: Something's not right here--I can only submit posts while the image picker is open, but with no image selected (this persists after the image is fully uploaded--GitHub is just stingy with video size):

https://github.com/user-attachments/assets/4ff2e31b-c01f-4878-a32e-6ece864f80b7

As far as I can tell it's `(imageManager?.progress ?? 1) != 1` that's causing the error, I threw some debug prints in and the other three states all update as expected.

File: Mlem/App/Views/Pages/PostEditor/PostEditorView+ImageView.swift [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+                        do {
+                            try await imageManager?.pasteFromClipboard(api: primaryApi)
+                        } catch {
+                            handleError(error)
+                        }
+                    }
+                }
+            }
+            .buttonStyle(ImageSourceButtonStyle())
+        }
```
Comment (EricBAndrews) [inline Mlem/App/Views/Pages/PostEditor/PostEditorView+ImageView.swift]:
```suggestion
                    }
                }
            }
            .font(.subheadline)
            .fontWeight(.semibold)
            .buttonStyle(ImageSourceButtonStyle())
        }
```

**Review by EricBAndrews — CHANGES_REQUESTED** [review]
Code looks good!

Noticed one logic issue, see below.

On the design side, the buttons look a little nicer to my eye with `.font(.subheadline)`, `.fontWeight(.semibold)` (here's a side-by-side for comparison):

Screenshot 2024-09-02 at 4 57 51 PM

I've put a comment below that applies those to all three buttons.

File: Mlem/App/Views/Pages/PostEditor/PostEditorView+LinkView.swift [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                        url = URL(string: pastedString, encodingInvalidCharacters: false)
+                    } else {
+                        return
+                    }
+                    if let url {
+                        link = .value(url)
+                    }
+                } label: {
+                    Label("Paste", systemImage: Icons.paste)
+                        .foregroundStyle(palette.selectedInteractionBarItem)
```
Comment (EricBAndrews) [inline Mlem/App/Views/Pages/PostEditor/PostEditorView+LinkView.swift]: For consistency with other suggestion about image buttons

```suggestion
                } label: {
                    Label("Paste", systemImage: Icons.paste)
                        .fontWeight(.semibold)
                        .foregroundStyle(palette.selectedInteractionBarItem)
```

**Review by EricBAndrews — CHANGES_REQUESTED** [review]

File: Mlem/App/Views/Pages/PostEditor/PostEditorView+ImageView.swift (hunk 2 above)
Comment (Sjmarf) [reply Mlem/App/Views/Pages/PostEditor/PostEditorView+ImageView.swift]: Yeah I like this 👍

File: Mlem/App/Views/Pages/PostEditor/PostEditorView+Logic.swift (hunk 1 above)
Comment (Sjmarf) [reply Mlem/App/Views/Pages/PostEditor/PostEditorView+Logic.swift]: Nice catch, fixed 👍

**Review by EricBAndrews — APPROVED** [review]
LGTM! 🔥

**Pull request merged**